### DISCUSSION TITLE
Allow searching for plural names and aliases in shopping list

### DISCUSSION THREAD

**Discussion by pyranha-ch** [mealie-recipes/mealie#3870] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

When quickly adding food items to the shopping list, you need to be very careful about what you type. For example, if you type "eggs" instead of "egg" you maybe won't find anything.

### Please provide a concise description of the feature that would resolve your issue.

When adding items to the shopping list by food, it only filters by the name. However it would be nice if it also filtered for the plural name, as well as for aliases.

### Please consider and list out some caveats or tradeoffs made in your design decision

-

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
I think this makes sense. FYI if the only reason you're trying to add the specific food to the shopping list is to make sure it gets labeled: "note" items are now automatically labeled based on your food store: https://github.com/mealie-recipes/mealie/pull/3800

For example, if you have an "egg" food item which has the "Refrigerated" label, and you add a note item "eggs" to your shopping list, it will get labeled with "Refrigerated" (even if you didn't actually select the "egg" food)

**Reply by pyranha-ch** [reply]
Great, didn't know that, even better! Thanks

**Comment by boc-the-git** [comment]
Backing up what Michael has said, we'd welcome a PR that adds this functionality, it makes perfect sense.
